packages/help-center/src/hooks/use-opening-coordinates.ts

```diff
@@ -0,0 +1,107 @@
+import { useMobileBreakpoint } from '@automattic/viewport-react';
+import { useState, useEffect } from 'react';
+
+const AESTHETIC_OFFSET = 20;
+
+/**
+ * This function calculates the position of the Help Center based on the last click event
+ *
+ * @param element The element that was clicked
+ * @returns object with left and top properties
+ */
+export const calculateOpeningPosition = ( element: HTMLElement ) => {
+	const { x, y, width, height } = element.getBoundingClientRect();
+	const { innerWidth, innerHeight } = window;
+	const helpCenterWidth = 410;
+	const helpCenterHeight = Math.min( 800, innerHeight * 0.8 );
+
+	const buttonLeftEdge = x;
+	const buttonRightEdge = x + width;
+
+	const buttonTopEdge = y;
+	const buttonBottomEdge = y + height;
+
+	const coords = { left: 0, top: 0, bottom: undefined, transformOrigin: `${ x }px ${ y }px` };
+
+	if ( buttonTopEdge + helpCenterHeight + AESTHETIC_OFFSET > innerHeight ) {
+		// Align the bottom edge of the help center with the top edge of the button
+		coords.top = buttonTopEdge - helpCenterHeight - AESTHETIC_OFFSET;
+		coords.transformOrigin = 'bottom';
+	} else {
+		// Align the top edge of the help center with the bottom edge of the button
+		coords.top = buttonBottomEdge + AESTHETIC_OFFSET;
+		coords.transformOrigin = 'top';
+	}
+
+	if ( buttonLeftEdge + helpCenterWidth + AESTHETIC_OFFSET > innerWidth ) {
+		// Align right edge of the help center with the right edge of the button
+		coords.left = buttonRightEdge - helpCenterWidth - AESTHETIC_OFFSET;
```

Comment:
Adding `AESTHETIC_OFFSET` here ruined the alignment with the button :(

It was so clean when they're aligned. I intentionally only added the gap vertically.